tests/test_invariants_present.py

```
@@ -153,3 +153,97 @@ def test_issue_template_present() -> None:
+
+
+def _dg309_h2_section_bodies(text: str) -> list[tuple[str, str]]:
+    """Return ``[(heading_line, body_text), ...]`` for each ``## `` (H2) section. Lines
+    inside fenced code blocks (``` or ~~~) are body, never headings, so a ``## `` inside a
+    fence cannot cause a false section split."""
+    sections: list[tuple[str, list[str]]] = []
+    in_fence = False
+    fence_marker = ""
+    for line in text.splitlines():
+        stripped = line.lstrip()
+        if stripped.startswith("```") or stripped.startswith("~~~"):
```

Ruff (0.15.15)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)